libvast/src/system/index.cpp

```diff
@@ -868,20 +868,25 @@ void index_state::decommission_active_partition(
                            partition_synopsis_pair{id, ps});
               unpersisted.erase(id);
               persisted_partitions.emplace(id);
+              self->send_exit(actor, caf::exit_reason::normal);
               if (completion)
                 completion(caf::none);
             },
             [=, this](const caf::error& err) {
               VAST_DEBUG("{} received error for request to persist partition "
                          "{} {}: {}",
                          *self, schema, id, err);
+              unpersisted.erase(id);
+              self->send_exit(actor, err);
```

Comment:
I don't think this is correct: If we fail to persist a partition it is still unpersisted. It makes sense to do this in the success case, but not so much in the error case.

Reply:
Doing nothing is not viable here. That would mean the partitions would stay in memory indefinitely.

Reply:
Should the debug message above then be on the error level and also say that it drops the partition? (Same for the other comments below)

Reply:
The one below was already an error. I clarified both messages.